client/starwhale/core/model/model.py

```diff
@@ -268,13 +269,19 @@ def eval_user_handler(
             else:
                 _scheduler.schedule_single_task(step, task_index)
         except Exception as e:
-            _manifest["status"] = STATUS.FAILED
+            _status = STATUS.FAILED
             _manifest["error_message"] = str(e)
             raise
         finally:
-            _status = True
-            for _step in _steps:
-                _status = _status and _step.status == STATUS.SUCCESS
+            _status = (
+                STATUS.SUCCESS
+                if all(
+                    _s.status == STATUS.SUCCESS
+                    for _s in _steps
+                    if (step and _s.step_name == step) or not step
```

Comment:
rename step -> step_name ?

Reply:
`if (step and _s.step_name == step) or not step` -> `if step == "" or step == _s.step_name` ?